irr/include/SSkinMeshBuffer.h

```diff
@@ -7,12 +7,153 @@
 #include "IMeshBuffer.h"
 #include "CVertexBuffer.h"
 #include "CIndexBuffer.h"
+#include "IVertexBuffer.h"
 #include "S3DVertex.h"
+#include "vector3d.h"
+#include <algorithm>
 #include <cassert>
+#include <memory>
+#include <numeric>
+#include <optional>
 
 namespace scene
 {
 
+struct WeightBuffer
+{
+	constexpr static u16 MAX_WEIGHTS_PER_VERTEX = 4;
+	size_t n_verts;
+	std::unique_ptr<u16[]> joint_idxs;
+	std::unique_ptr<f32[]> weights;
```

Comment:
any reason not use `std::vector` for better array safety here?

Reply:
Counter-Proposal: since those two arrays always come in pairs - how about using a struct instead?
```C++
struct VertexData {
	std::array<u16, MAX_WEIGHTS_PER_VERTEX> joint_indices;
	std::array<u32, MAX_WEIGHTS_PER_VERTEX> weights;
};

std::vector<VertexData> data;

// in ctor
	data.resize(n_verts);
	// this->n_verts is now found in data.size()`, and each array can be traversed trivially  using a for-loop
```


FWIW `sizeof(std::array<uint16_t, 10>)` evaluates to `20`, thus no memory overhead.